### PR TITLE
✨ feat: Add BurnerHandler to create and manage burner accounts

### PR DESCRIPTION
**Pull request by Cheelax**
This commit adds the `BurnerHandler` class to the `create-burner` package. The `BurnerHandler` is responsible for creating and managing burner accounts. It includes methods for initializing, listing, selecting, getting, creating, and clearing burner accounts.

The `BurnerHandler` constructor takes a configuration object that includes the necessary parameters such as the node URL, master address, master private key, and account class hash.

The `initialize` method initializes the burner manager by creating the burner accounts and performing initialization tasks. If an error occurs during initialization, it is logged and thrown.

The `listBurner` method returns a list of all burner accounts managed by the `BurnerManager`.

The `selectBurner` method selects a burner account based on the given address and returns the active account.

The `getBurner` method retrieves a burner account based on the given address.

The `createBurnerAccount` method creates a new burner account using the `burnerManager` and returns the created account.

The `clearBurner` method clears all burner accounts managed by the `BurnerManager`.

Additionally, this commit exports the `BurnerHandler` class from the `manager/index.ts` file, allowing it to be used outside the module alongside `BurnerManager` and `prefundAccount`.

### CODE REVIEW

File: packages/create-burner/src/manager/burnerHandler.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return this.burnerManager.getActiveAccount();
+    };
+    
+    getBurner = (address: string) => {
+        return this.burnerManager.get(address);
+    };
+    
+    createBurnerAccount = async (burnerManager: string) => {
+        const newAccount = await this.burnerManager.create();
+        return newAccount;
```
Comment (ponderingdemocritus) [inline packages/create-burner/src/manager/burnerHandler.ts:56]:
```suggestion
        return await this.burnerManager.create();;
```

**Comment by rsodre** [comment]
Nice!!

Might be a good idea to add it to the examples.

**Comment by Cheelax** [comment]
Ok, handler is working in zConqueror.
Here is how I implement it there in my dojoContext:
`import { createContext, ReactNode, useContext, useMemo, useState, useEffect } from 'react';
import { SetupResult } from './dojo/setup';
import { BurnerHandler } from '@dojoengine/create-burner';
import { Account, RpcProvider } from 'starknet';

type Context = {
  setup: SetupResult;
  account: {
    create: () => Promise<void>;
    list: () => any[];
    get: (id: string) => Account | undefined;
    select: (id: string) => void;
    account: Account | undefined;
    masterAccount: Account;
    isDeploying: boolean;
    clear: () => void;
  };
};

const DojoContext = createContext<Context | null>(null);

type Props = {
  children: ReactNode;
  value: SetupResult;
};

export const DojoProvider = ({ children, value }: Props) => {
  const [burnerHandler, setBurnerHandler] = useState<BurnerHandler>();
  const [isDeploying, setIsDeploying] = useState(false);

  const [selectedAccountId, setSelectedAccountId] = useState<string | undefined>(undefined);

  useEffect(() => {
    const config = {
      nodeUrl: import.meta.env.VITE_PUBLIC_NODE_URL!,
      masterAddress: import.meta.env.VITE_PUBLIC_MASTER_ADDRESS!,
      masterPrivateKey: import.meta.env.VITE_PUBLIC_MASTER_PRIVATE_KEY!,
      accountClassHash: import.meta.env.VITE_PUBLIC_ACCOUNT_CLASS_HASH!,
    };

    const handler = new BurnerHandler(config);
    handler.initialize().then(() => {
      console.log('BurnerHandler initialized');
      setBurnerHandler(handler);
    });
  }, []);

  const provider = useMemo(
    () =>
      new RpcProvider({
        nodeUrl: import.meta.env.VITE_PUBLIC_NODE_URL!,
      }),
    [import.meta.env.VITE_PUBLIC_NODE_URL!]
  );

  const masterAddress = import.meta.env.VITE_PUBLIC_MASTER_ADDRESS!;
  const privateKey = import.meta.env.VITE_PUBLIC_MASTER_PRIVATE_KEY!;

  useEffect(() => {
    if (burnerHandler) {
      setSelectedAccountId(burnerHandler.getActiveAccount()?.address);
    }
  }, [burnerHandler]);

  const masterAccount = useMemo(
    () => new Account(provider, masterAddress, privateKey),
    [provider, masterAddress, privateKey]
  );

  const selectedAccount = useMemo(() => {
    if (selectedAccountId) return burnerHandler?.getBurner(selectedAccountId);
  }, [burnerHandler, selectedAccountId]);

  const contextValue: Context = {
    setup: value,
    account: {
      create: async () => {
        setIsDeploying(true);
        await burnerHandler?.createBurnerAccount();
        setIsDeploying(false);
      },
      list: () => burnerHandler?.listBurner() || [],
      get: (id: string) => burnerHandler?.getBurner(id),
      select: (id: string) => {
        burnerHandler?.selectBurner(id);
        setSelectedAccountId(burnerHandler?.getActiveAccount()?.address);
      },
      account: selectedAccount || undefined,
      masterAccount,
      isDeploying,
      clear: () => burnerHandler?.clearBurner(),
    },
  };

  return <DojoContext.Provider value={contextValue}>{children}</DojoContext.Provider>;
};

export const useDojo = () => {
  const value = useContext(DojoContext);
  if (!value) throw new Error('Must be used within a DojoProvider');
  return value;
};
`

Do you want me to improve some stuff still? @ponderingdemocritus

**Comment by ponderingdemocritus** [comment]
Yeah this is looking good - could you try work this into the phaser example? Have a look at emoji-man for inspo

**Comment by ponderingdemocritus** [comment]
Im thinking a better pattern is to pass the burnerManager into the useBurnerManager hook. The above repeats a bunch of code. Going to close for now.

**Pull request closed without merging**